### PR TITLE
fix: allow slashes in weave ref object ids

### PR DESCRIPTION
**Pull request by jamie-rasmussen**
This changes UI side ref parsing; I believe we still have some related problems in backend code.

Internal Jira:
https://wandb.atlassian.net/browse/WB-18522
https://wandb.atlassian.net/browse/WB-18743

### CODE REVIEW

File: weave-js/src/react.tsx [hunk 1]
```diff
@@ -499,6 +501,39 @@ export const isWeaveObjectRef = (ref: ObjectRef): ref is WeaveObjectRef => {
...
+const RE_WEAVE_OBJECT_REF_PATHNAME = new RegExp(
+  [
+    '^', // Start of the string
+    PATTERN_ENTITY,
+    '/',
+    PATTERN_PROJECT,
+    '/',
+    '(object|op)', // Weave kind
+    '/',
+    '([a-zA-Z0-9-_/. ]{1,128})', // Artifact name
```
Comment (jamie-rasmussen) [inline weave-js/src/react.tsx:515]: Would like to draw attention to this.

The wandb SDK limits artifact names.
https://github.com/wandb/wandb/blob/main/wandb/sdk/artifacts/artifact.py#L101

> Names can contain letters, numbers, underscores, hyphens, and dots.


The W&B backend is not as restrictive. It just checks <= 128 chars and allows everything but ‘/' and ‘:’ Last I checked the artifacts UI does not even validate against that. For this reason, there might be artifacts (such as Launch Jobs) whose names contain characters such as spaces that you could not create through the public W&B SDK API.

In Weave multiple external users have now published with '/' and at least one internal user has used spaces. I think we may want to be more restrictive at publish time, but do we want to allow these now to unbreak the users?

Comment (tssweeney) [reply weave-js/src/react.tsx:515]: Personally, I would rather avoid slashes entirely and stick to a more conservative dictionary for now. It feels better to intentionally add support, than have to deal with accidental ramifications.

That being said, now that you have handled this case, maybe there isn't a problem with including support for these symbols?

At the very least, we should certainly add the same restriction to the backend (and write client) to protect users from including other symbols / emojies / etc.

**Review by tssweeney — APPROVED** [review]

File: weave-js/src/react.tsx (hunk 1 above)
Comment (jamie-rasmussen) [reply weave-js/src/react.tsx:515]: We never really resolved the conversation around Python API design and whether it was OK to raise on an invalid name. https://weightsandbiases.slack.com/archives/C03BSTEBD7F/p1714676345298539

Comment (jamie-rasmussen) [reply weave-js/src/react.tsx:515]: Also we still need to discuss the pros/cons of validating at the pydantic level: https://github.com/wandb/weave/pull/1597

**Pull request merged**